### PR TITLE
Add layer markdown

### PR DESCRIPTION
**Pull request by jenniew**
Add docs.
Containers
    ConcatTable
    MapTable
    ParallelTable

Functional API
    CMul
    MaskedSelect
    Power
    Squeeze

Loss Functions
    CrossEntropyCriterion
    TimeDistributedCriterion

Non-linear Activations
    Sigmoid

Normalization Layers
    SpatialBatchNormalization

optim
    Adam

### CODE REVIEW

File: docs/docs/APIdocs/layers/nn/Containers/MapTable.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+## MapTable ##
+
+**Scala:**
+```scala
+val mod = MapTable(module)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Containers/MapTable.md]: better add the default value of argument if any. e.g.
```scala
val mod = MapTable(module=null)
```

File: docs/docs/APIdocs/layers/nn/Containers/MapTable.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+```
+**Python:**
+```python
+mod = MapTable(module)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Containers/MapTable.md]: default parameter value for `module`

File: docs/docs/APIdocs/layers/nn/Containers/MapTable.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+`module` a member module.  
+ 
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.utils.T 
+
+val map = MapTable[Float]()
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Containers/MapTable.md]: we expect user to use our API without [T] e.g. 
```scala
import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat
val map = MapTable()
```

File: docs/docs/APIdocs/layers/nn/Containers/ParallelTable.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+It is a container module that applies the i-th member module to the i-th
+ input, and outputs an output in the form of Table
+ 
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.utils.T 
+
+val module = ParallelTable[Double]()
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Containers/ParallelTable.md]: [T]

File: docs/docs/APIdocs/layers/nn/Functional API/CMul.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
+## CMul ##
+
+**Scala:**
+```scala
+val module = CMul(size)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Functional API/CMul.md]: missing parameter, and specify default value
```scala
CMul(size: Array[Int], wRegularizer: Regularizer[T] = null)
```

File: docs/docs/APIdocs/layers/nn/Functional API/MaskedSelect.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+ 
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.utils.T
+import scala.util.Random
+
+
+val layer = MaskedSelect[Double]()
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Functional API/MaskedSelect.md]: [T]

File: docs/docs/APIdocs/layers/nn/Functional API/Power.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
+## Power ##
+
+**Scala:**
+```scala
+val module = Power(power, scale, shift)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Functional API/Power.md]: specify default value of parameters if any.

File: docs/docs/APIdocs/layers/nn/Normalization Layers/SpatialBatchNormalization.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
+## SpatialBatchNormalization ##
+
+**Scala:**
+```scala
+val module = SpatialBatchNormalization(nOutput, eps, momentum, affine,
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Normalization Layers/SpatialBatchNormalization.md]: default values

File: docs/docs/APIdocs/layers/optim/Adam.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
+## Adam ##
+
+**Scala:**
+```scala
+val optim = Adam(learningRate, learningRateDecay, beta1, beta2, Epsilon)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/optim/Adam.md]: default values

File: docs/docs/APIdocs/layers/optim/Adam.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+    (fout, dxout)
+  }
+val x = Tensor[Double](2).fill(0)
+val config = T("learningRate" -> 0.002)
+> print(optm.optimize(rosenBrock, x, config))
+(0.001999999683772284
+0.0
+[com.intel.analytics.bigdl.tensor.DenseTensor$mcD$sp of size 2],[D@302d88d8)
+```
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/optim/Adam.md]: missing python example

File: docs/docs/APIdocs/layers/nn/Containers/ConcatTable.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+ConcateTable usually works with CAddTable and CMulTable to
+ implement element wise add/multiply on outputs of two modules.
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.Tensor
+
+val mlp = ConcatTable[Activity, Float]()
```
Comment (jenniew) [inline docs/docs/APIdocs/layers/nn/Containers/ConcatTable.md]: Updated

File: docs/docs/APIdocs/layers/nn/Containers/MapTable.md (hunk 1 above)
Comment (jenniew) [inline docs/docs/APIdocs/layers/nn/Containers/MapTable.md]: Updated

File: docs/docs/APIdocs/layers/nn/Containers/MapTable.md (hunk 2 above)
Comment (jenniew) [inline docs/docs/APIdocs/layers/nn/Containers/MapTable.md]: Updated

File: docs/docs/APIdocs/layers/nn/Containers/MapTable.md (hunk 3 above)
Comment (jenniew) [inline docs/docs/APIdocs/layers/nn/Containers/MapTable.md]: Updated

File: docs/docs/APIdocs/layers/nn/Containers/ParallelTable.md (hunk 4 above)
Comment (jenniew) [inline docs/docs/APIdocs/layers/nn/Containers/ParallelTable.md]: Updated

File: docs/docs/APIdocs/layers/nn/Functional API/CMul.md (hunk 5 above)
Comment (jenniew) [inline docs/docs/APIdocs/layers/nn/Functional API/CMul.md]: Updated

File: docs/docs/APIdocs/layers/nn/Functional API/MaskedSelect.md (hunk 6 above)
Comment (jenniew) [inline docs/docs/APIdocs/layers/nn/Functional API/MaskedSelect.md]: Updated

File: docs/docs/APIdocs/layers/nn/Functional API/Power.md (hunk 7 above)
Comment (jenniew) [inline docs/docs/APIdocs/layers/nn/Functional API/Power.md]: Updated

File: docs/docs/APIdocs/layers/nn/Normalization Layers/SpatialBatchNormalization.md (hunk 8 above)
Comment (jenniew) [inline docs/docs/APIdocs/layers/nn/Normalization Layers/SpatialBatchNormalization.md]: Updated

File: docs/docs/APIdocs/layers/optim/Adam.md (hunk 9 above)
Comment (jenniew) [inline docs/docs/APIdocs/layers/optim/Adam.md]: Updated

File: docs/docs/APIdocs/layers/optim/Adam.md (hunk 10 above)
Comment (jenniew) [inline docs/docs/APIdocs/layers/optim/Adam.md]: Updated

File: docs/docs/APIdocs/layers/nn/Containers/ConcatTable.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+```scala
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat
+
+val mlp = ConcatTable()
+mlp.add(Linear(3, 2))
+mlp.add(Linear(3, 4))
+
+> print(mlp.forward(Tensor[Float](2, 3).rand()))
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Containers/ConcatTable.md]: Remove `[Float]`

File: docs/docs/APIdocs/layers/nn/Containers/MapTable.md [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+```scala
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.utils.T 
+
+val map = MapTable()
+map.add(Linear(10, 3))
+val input = T(
+      Tensor[Float](10).randn(),
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Containers/MapTable.md]: Remove `[Float]`

File: docs/docs/APIdocs/layers/nn/Non-linear Activations/Sigmoid.md [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+
+Sigmoid is defined as: f(x) = 1 / (1 + exp(-x))
+  
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.tensor.Tensor
+
+val layer = new Sigmoid[Float]()
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Non-linear Activations/Sigmoid.md]: remove [Float]
